allennlp/data/dataset_readers/atis.py

```diff
@@ -0,0 +1,122 @@
+import json
+from typing import Dict, List
+import logging
+
+from overrides import overrides
+from parsimonious.exceptions import ParseError
+
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, ArrayField, ListField, IndexField, \
+        ProductionRuleField, TextField, MetadataField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import SingleIdTokenIndexer, TokenIndexer
+from allennlp.data.tokenizers import Tokenizer, WordTokenizer
+from allennlp.data.tokenizers.word_splitter import SpacyWordSplitter
+
+from allennlp.semparse.worlds.atis_world import AtisWorld
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+def lazy_parse(text: str):
+    for interaction in text.split("\n"):
+        if interaction:
+            yield json.loads(interaction)
+
+@DatasetReader.register("atis")
+class AtisDatasetReader(DatasetReader):
+    """
+    This ``DatasetReader`` takes json files and converts them into ``Instances`` for the
+    ``AtisSemanticParser``.
+    """
+    def __init__(self,
+                 source_token_indexers: Dict[str, TokenIndexer] = None,
+                 target_token_indexers: Dict[str, TokenIndexer] = None,
+                 token_indexers: Dict[str, TokenIndexer] = None,
+                 lazy: bool = False,
+                 tokenizer: Tokenizer = None
+                ) -> None:
+        super().__init__(lazy)
+        self._token_indexers = token_indexers or {'tokens': SingleIdTokenIndexer()}
+        self._tokenizer = tokenizer or WordTokenizer(SpacyWordSplitter(pos_tags=True))
+
+        self._source_token_indexers = source_token_indexers or {"tokens": SingleIdTokenIndexer()}
+        self._target_token_indexers = target_token_indexers or self._source_token_indexers
+
+
+    @overrides
+    def _read(self, file_path: str):
+        # if `file_path` is a URL, redirect to the cache
+        file_path = cached_path(file_path)
+
+        with open(file_path) as atis_file:
+            logger.info("Reading ATIS instances from dataset at : %s", file_path)
+            for line in lazy_parse(atis_file.read()):
+                utterances = []
+                for current_interaction in line['interaction']:
+                    nl_key = 'utterance' if 'utterance' in current_interaction else 'nl_with_dates'
+
+                    utterances.append(current_interaction[nl_key])
+                    world = AtisWorld(utterances)
+
+                    try:
+                        action_sequence = world.get_action_sequence(current_interaction['sql'])
+                    except ParseError:
```

Comment:
The `ParseError` is thrown in the `parsimonious` code, I think we need it because we need some kind of error handling when some query doesn't parse or else it will break. I'm not sure what the alternative for not having this check is, what do you think?

Reply:
Ah, sorry, I was thinking of a [`ParsingError`](https://github.com/allenai/allennlp/blob/12b74e5a91ef52bea2c1574fc76b52e8f78f86ac/allennlp/semparse/worlds/world.py#L15).  If `parsimonious` will throw this, then yes, you should definitely catch it here.